src/Rector/Dynamic/ReturnTypehintRector.php

```diff
@@ -0,0 +1,171 @@
+<?php declare(strict_types=1);
+
+namespace Rector\Rector\Dynamic;
+
+use PhpParser\Node;
+use PhpParser\Node\Identifier;
+use PhpParser\Node\Name\FullyQualified;
+use PhpParser\Node\NullableType;
+use PhpParser\Node\Param;
+use PhpParser\Node\Stmt\Class_;
+use PhpParser\Node\Stmt\ClassLike;
+use PhpParser\Node\Stmt\ClassMethod;
+use Rector\Node\Attribute;
+use Rector\NodeTypeResolver\NodeTypeResolver;
+use Rector\Php\TypeAnalyzer;
+use Rector\Rector\AbstractRector;
+use Rector\RectorDefinition\CodeSample;
+use Rector\RectorDefinition\RectorDefinition;
+
+final class ReturnTypehintRector extends AbstractRector
+{
+    /**
+     * class => [
+     *      method => typehting
+     * ]
+     *
+     * @var string[][]
+     */
+    private $typehintForMethodByClass = [];
+
+    /**
+     * @var NodeTypeResolver
+     */
+    private $nodeTypeResolver;
+
+    /**
+     * @var TypeAnalyzer
+     */
+    private $typeAnalyzer;
+
+    /**
+     * @param mixed[] $typehintForMethodByClass
+     */
+    public function __construct(
+        array $typehintForMethodByClass,
+        NodeTypeResolver $nodeTypeResolver,
+        TypeAnalyzer $typeAnalyzer
+    ) {
+        $this->typehintForMethodByClass = $typehintForMethodByClass;
+        $this->nodeTypeResolver = $nodeTypeResolver;
+        $this->typeAnalyzer = $typeAnalyzer;
+    }
+
+    public function getDefinition(): RectorDefinition
+    {
+        return new RectorDefinition('[Dynamic] Changes defined return typehint of method and class.', [
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+class SomeClass
+{
+    public getData();
```

Comment:
Missing dockblock or it will be set in the `rector.yml`?

Reply:
This rector completes typehints only. Or what do you mean?

Reply:
I meant if this Rector will discover the typehint, or will be provided :smile:

Reply:
Provided :) It should be clear from tests, isn't it?

Rectors always works with prepared configuration, so they're reliable in this matter. 

Deduction can be pre-process of creating such configs, but that's another story